inc/plugin.class.php

```diff
@@ -360,17 +360,32 @@ static function loadLang($plugin_key, $forcelang = '', $coretrytoload = '') {
          );
       }
 
-      $mofile = str_replace($locales_dir, GLPI_LOCAL_I18N_DIR . '/'.$plugin_key . '/', $mofile);
-      $phpfile = str_replace('.mo', '.php', $mofile);
+      $plugin_folders = scandir(GLPI_LOCAL_I18N_DIR);
+      $plugin_folders = array_filter($plugin_folders, function($dir) use ($plugin_key) {
+         if (!is_dir(GLPI_LOCAL_I18N_DIR . "/$dir")) {
+            return false;
+         }
 
-      // Load local PHP file if it exists
-      if (file_exists($phpfile)) {
-         $TRANSLATE->addTranslationFile('phparray', $phpfile, $plugin_key, $coretrytoload);
-      }
+         if ($dir == $plugin_key) {
+            return true;
+         }
 
-      // Load local MO file if it exists -- keep last so it gets precedence
-      if (file_exists($mofile)) {
-         $TRANSLATE->addTranslationFile('gettext', $mofile, $plugin_key, $coretrytoload);
+         return strpos($dir, $plugin_key . '_') !== false;
```

Comment:
Using a regex would be more secure.
For instance, when loading `dashboard` plugin locales, we do not want to load `mydashboard` and `advanceddashboard` custom locales.
```php
return preg_match('/^' . preg_quote($plugin_key, '/') . '(_.+)?$'/', $dir);
```